### PR TITLE
Fixes #9616 remove kubectl version check

### LINKED ISSUE #9616: Remove kubectl version check from linkerd check
The `linkerd check` command [verifies that kubectl meets the version found](https://github.com/linkerd/linkerd2/blob/c492bdb290359b6ffb665a813dc1503c45c9ad5d/pkg/healthcheck/healthcheck.go#L542-L549) in [pkg/k8s/api.go](https://github.com/linkerd/linkerd2/blob/c492bdb290359b6ffb665a813dc1503c45c9ad5d/pkg/k8s/api.go#L30). This check is no longer necessary because the CLI no longer exec's `kubectl`.

We want to remove the check and any dead code that results.

Related to Website issue [#1510](https://github.com/linkerd/website/issues/1510)

### PR DESCRIPTION
**Pull request by ziollek**
Signed-off-by: tomasz.ziolkowski <e.prace@gmail.com>

Fixes #9616

### CODE REVIEW

**Review by awmackowiak — APPROVED** [review]

**Review by smigacz — APPROVED** [review]

File: pkg/k8s/k8s.go [hunk 1]
```diff
@@ -1,232 +0,0 @@
-package k8s
```
Comment (mateiidavid) [inline pkg/k8s/k8s.go:1]: Why has this entire file been removed? A lot of the stuff inside is used in `viz`

**Review by mateiidavid — CHANGES_REQUESTED** [review]

File: pkg/k8s/k8s.go (hunk 1 above)
Comment (ziollek) [reply pkg/k8s/k8s.go]: Thanks for catching this, I didn't know how I could mistake that. Obviously kubectl.go was supposed to be deleted. It was only used in outdated healthcheck.

**Review by mateiidavid — APPROVED** [review]
@ziollek thanks a lot for the PR, and for the quick turnaround with the requested changes! This looks good to ship 🚢

**Review by alpeb — APPROVED** [review]
LGTM :+1: 
Also fixes #9615

**Pull request merged**